blog/services/pin_service.py

```
@@ -0,0 +1,98 @@
+
+    current_count = pinned_qs.count()
+    if current_count >= max_count:
+        raise PinLimitExceededError()
+
+    last_sort_order = pinned_qs.aggregate(max_sort=Max("sort_order"))["max_sort"] or 0
+
+    try:
+        Pin.objects.create(post=post, sort_order=last_sort_order + 1)
+    except IntegrityError:
+        # 동시성이나 데이터 이상 상황 방어
+        raise AlreadyPinnedError()
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)